ui/src/app/pages/data/concept/concept-search.tsx

```diff
@@ -120,302 +121,305 @@ interface State {
+  withCurrentCohortSearchContext(),
+  withCurrentConcept(),
+  withCurrentWorkspace(),
+  withUrlParams(),
+  withNavigation)
+(class extends React.Component<Props, State> {
+  subscription: Subscription;
+  constructor(props: any) {
+    super(props);
+    this.state = {
+      copying: false,
+      copySaving: false,
+      conceptSet: undefined,
+      editing: false,
+      editName: '',
+      editDescription: '',
+      editSaving: false,
+      error: false,
+      errorMessage: '',
+      deleting: false,
+      loading: this.isDetailPage,
+      showMoreDescription: false,
+      unsavedChanges: false,
+      conceptSetUpdating: false
+    };
+  }
 
-    componentDidMount() {
-      this.props.hideSpinner();
-      if (this.isDetailPage) {
-        this.getConceptSet();
-      } else if (!currentConceptStore.getValue()) {
-        currentConceptStore.next([]);
-      }
-      this.subscription = currentConceptStore.subscribe(currentConcepts => {
-        if (![null, undefined].includes(currentConcepts)) {
-          this.checkUnsavedConceptChanges(currentConcepts);
-        }
-      });
-      this.subscription.add(conceptSetUpdating.subscribe(updating => this.setState({conceptSetUpdating: updating})));
-      navigationGuardStore.set({component: this});
-    }
-
-    componentWillUnmount() {
-      localStorage.removeItem(LOCAL_STORAGE_KEY_CRITERIA_SELECTIONS);
-      currentConceptStore.next(undefined);
-      currentConceptSetStore.next(undefined);
-      this.subscription.unsubscribe();
-      navigationGuardStore.set(null);
-    }
+  componentDidMount() {
+    this.props.hideSpinner();
 
-    checkUnsavedConceptChanges(currentConcepts) {
-      const currentConceptSet = currentConceptSetStore.getValue();
-      const currentConceptsMap = currentConcepts.map(concept => {
-        ['attributes', 'parameterId'].forEach(prop => delete concept[prop]);
-        return concept;
-      });
-      const unsavedChanges = (!currentConceptSet && currentConcepts.length > 0) ||
-        (!!currentConceptSet && sortAndStringify(currentConceptSet.criteriums) !== sortAndStringify(currentConceptsMap));
-      this.setState({unsavedChanges});
+    if (!this.isDetailPage && !currentConceptStore.getValue()) {
+      currentConceptStore.next([]);
     }
+    this.subscription = currentConceptStore.subscribe(currentConcepts => {
+      if (![null, undefined].includes(currentConcepts)) {
+        this.checkUnsavedConceptChanges(currentConcepts);
+      }
+    });
+    this.subscription.add(conceptSetUpdating.subscribe(updating => this.setState({conceptSetUpdating: updating})));
+  }
 
-    async getConceptSet() {
-      const {urlParams: {ns, wsid, csid}} = this.props;
-      try {
-        const resp = await conceptSetsApi().getConceptSet(ns, wsid, csid);
-        if (resp.domain === Domain.SURVEY) {
-          resp.criteriums = resp.criteriums.filter((survey) => survey.parentCount !== 0);
+  componentDidUpdate(prevProps, prevState) {
```

Comment:
This logic (this component) is complicated and it doesn't need to be. Here's what it's trying to do and what we should do to make this not necessary:

- This component does two and a half things - it allows users to search concepts in various domains, and it allows users to view preexisting concept sets. (The half is that you can add concepts to said preexisting concept sets.)
- `isDetailPage` is the thing that determines which of these things it is doing at any given point in time. That is determined by `urlParams.csid`.
- For some reason, we store the concept set retrieved from `urlParams.csid` in `currentConceptSetStore` as well as in local state (probably because you can go search more concepts to add for the concept set.)
- There's a bug of sorts with our router nesting where we update `urlParamsStore` in a HOC around each route that doesn't execute until after the component mounts, so `urlParams.csid` doesn't exist when this component first renders even if the route includes a `csid`.
- So, this `componentDidUpdate` is meant to load the concept set once we get it.

There's several ways this could be done better, but they all involve tearing the component out.